src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,159 @@
+import { Link } from 'react-router-dom';
+import './Carousel.scss';
+
+const Carousel = () => {
+  const mockProducts = [
+    {
+      id: 1,
+      imgURL:
+        'https://images.unsplash.com/photo-1543508282-6319a3e2621f?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8OXx8c2hvZXN8ZW58MHx8MHx8&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 2,
+      imgURL:
+        'https://images.unsplash.com/photo-1600185365483-26d7a4cc7519?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTh8fHNob2VzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 3,
+      imgURL:
+        'https://images.unsplash.com/photo-1611510338559-2f463335092c?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8OTB8fHNob2VzfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 4,
+      imgURL:
+        'https://images.unsplash.com/photo-1605348532760-6753d2c43329?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MjJ8fG5pa2V8ZW58MHx8MHx8&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 5,
+      imgURL:
+        'https://images.unsplash.com/photo-1605408499391-6368c628ef42?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NDN8fG5pa2V8ZW58MHx8MHx8&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 6,
+      imgURL:
+        'https://images.unsplash.com/photo-1579338908476-3a3a1d71a706?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8ODF8fG5pa2V8ZW58MHx8MHx8&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 7,
+      imgURL:
+        'https://images.unsplash.com/photo-1602231379593-b85a472e3c99?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTU0fHxuaWtlfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 8,
+      imgURL:
+        'https://images.unsplash.com/photo-1549298916-b41d501d3772?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTc0fHxuaWtlfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+    {
+      id: 9,
+      imgURL:
+        'https://images.unsplash.com/photo-1626379616459-b2ce1d9decbc?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NTc0fHxuaWtlfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=60',
+      name: '나이키 에어포스',
+      gender: '남성신발',
+      price: '99000',
+    },
+  ];
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const [style, setStyle] = useState({
+    transform: `translateX(-${currentSlide * 100}%)`,
+    transition: 'all 0.3s ease-out',
+  });
```

Comment:
style은 currentSlide 의 computed 관계이기 떄문에 
```suggestion
  const style = {
    transform: `translateX(-${currentSlide * 100}%)`,
    transition: 'all 0.3s ease-out',
  };
```

한번 console.log('Carousel Render!!'); 를 찍어보셔서 얼마나 재 렌더링이 줄어드는지 확인해주세요!

Reply:
재 렌더링을 막는 이유는 렌더링이 퍼포먼스를 가장 많이 깎아먹는 요소이기 때문입니다.
그래서 최적화를 위해 불필요한 
state, effect 줄이기 => memo => useMemo or useCallback 순서로 퍼포먼스 개선 합니다.

여기서 알려드린건 state, effect 줄이기입니다!

Reply:
수정 전과 수정 후의 next, prev 버튼 눌렀을때의 콘솔도 확인해보시면 더 좋을 것 같습니다. 👍🏻